docs/current_docs/manuals/developer/cookbook.mdx

```diff
@@ -698,3 +698,59 @@ Set environment variables in a container:
 ```shell
 dagger call set-env
 ```
+
+## Error handling
+
+### Terminate gracefully
+
+The following Dagger Function demonstrates how to handle errors in a pipeline.
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go file=./cookbook/snippets/handle-errors/go/main.go
+```
+
+</TabItem>
+<TabItem value="Python">
+
+```python file=./cookbook/snippets/handle-errors/python/main.py
+```
+
+</TabItem>
+<TabItem value="TypeScript">
+
+```typescript file=./cookbook/snippets/handle-errors/typescript/index.ts
+```
+
+</TabItem>
+</Tabs>
+
+### Continue using a container after command execution fails
+
+The following Dagger Function demonstrates how to continue using a container after a command executed within it fails. A common use case for this is to export a report that a test suite tool generates.
+
+:::note
+The caveat with this approach is that forcing a zero exit code on a failure caches the failure. This may not be desired depending on the use case.
+:::
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go file=./cookbook/snippets/continue-after-errors/go/main.go
```

Comment:
These snippets (all 3) use file.export to "save the report locally" which will not work the same with functions. If we want to save the file, the function should return a `File`. But then we can't print `Tests Passed/failed`

Reply:
Agreed live to return a custom type. Will follow up with example